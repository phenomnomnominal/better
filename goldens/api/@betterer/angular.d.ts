export declare function angular(configFilePath: string, extraCompilerOptions: CompilerOptions): BettererFileTest;
