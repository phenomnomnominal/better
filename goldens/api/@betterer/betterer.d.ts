export declare function betterer(partialConfig?: BettererStartConfigPartial): Promise<BettererSummary>;
export declare namespace betterer {
    var file: typeof import("./betterer").file;
    var watch: typeof import("./betterer").watch;
}

export declare type BettererBaseConfigPartial = Partial<{
    configPaths: BettererConfigPaths | string;
    cwd: string;
    filters: ReadonlyArray<string | RegExp> | string;
    reporters: BettererConfigReporters;
    resultsPath: string;
    silent: boolean;
    tsconfigPath: string;
}>;

export declare type BettererConfig = {
    allowDiff: boolean;
    configPaths: BettererConfigPaths;
    cwd: string;
    filters: BettererConfigFilters;
    ignores: BettererConfigIgnores;
    reporters: BettererConfigReporters;
    resultsPath: string;
    silent: boolean;
    tsconfigPath: string | null;
    update: boolean;
};

export declare type BettererConfigFilters = ReadonlyArray<RegExp>;

export declare type BettererConfigIgnores = ReadonlyArray<string>;

export declare type BettererConfigPartial = BettererBaseConfigPartial & BettererStartConfigPartial & BettererWatchConfigPartial;

export declare type BettererConfigPaths = ReadonlyArray<string>;

export declare type BettererConfigReporters = ReadonlyArray<BettererConfigReporter>;

export declare type BettererContext = {
    readonly config: BettererConfig;
    readonly lifecycle: Promise<BettererSummaries>;
};

export declare type BettererDeserialise<DeserialisedType extends BettererResultValue, SerialisedType> = (serialised: SerialisedType) => DeserialisedType;

export declare type BettererDiff<DeserialisedType extends BettererResultValue = BettererResultValue, DiffType = null> = {
    expected: DeserialisedType;
    result: DeserialisedType;
    diff: DiffType;
    log: (logger: BettererLogger) => void;
};

export declare type BettererDiffer<DeserialisedType extends BettererResultValue, DiffType> = (expected: DeserialisedType, result: DeserialisedType) => BettererDiff<DeserialisedType, DiffType>;

export declare type BettererFile = BettererFileBase & {
    addIssue(start: number, end: number, message: string, hash?: string): void;
    addIssue(line: number, col: number, length: number, message: string, hash?: string): void;
    addIssue(startLine: number, startCol: number, endLine: number, endCol: number, message: string, hash?: string): void;
};

export declare type BettererFileBase = {
    readonly absolutePath: string;
    readonly hash: string;
    readonly issues: BettererFileIssues;
    readonly key: string;
};

export declare type BettererFileDiff = {
    fixed?: BettererFileIssues;
    existing?: BettererFileIssues;
    new?: BettererFileIssues;
};

export declare type BettererFileGlobs = ReadonlyArray<string | ReadonlyArray<string>>;

export declare type BettererFileIssue = {
    readonly line: number;
    readonly column: number;
    readonly length: number;
    readonly message: string;
    readonly hash: string;
};

export declare type BettererFileIssues = ReadonlyArray<BettererFileIssue>;

export declare type BettererFilePaths = ReadonlyArray<string>;

export declare type BettererFilePatterns = ReadonlyArray<RegExp | ReadonlyArray<RegExp>>;

export declare class BettererFileResolver {
    get cwd(): string;
    constructor(depth?: number);
    excludeΔ(...excludePatterns: BettererFilePatterns): this;
    files(filePaths: BettererFilePaths): Promise<BettererFilePaths>;
    includeΔ(...includePatterns: BettererFileGlobs): this;
    resolve(...pathSegments: Array<string>): string;
    validate(filePaths: BettererFilePaths): Promise<BettererFilePaths>;
}

export declare class BettererFileTest implements BettererTestBase<BettererFileTestResult, BettererFileIssuesMapSerialised, BettererFilesDiff> {
    get config(): BettererTestConfig<BettererFileTestResult, BettererFileIssuesMapSerialised, BettererFilesDiff>;
    readonly isBettererFileTest = "isBettererFileTest";
    readonly isBettererTest = "isBettererTest";
    get isOnly(): boolean;
    get isSkipped(): boolean;
    constructor(_resolver: BettererFileResolver, fileTest: BettererFileTestFunction);
    exclude(...excludePatterns: BettererFilePatterns): this;
    include(...includePatterns: BettererFileGlobs): this;
    only(): this;
    skip(): this;
}

export declare type BettererFileTestDiff = BettererDiff<BettererFileTestResult, BettererFilesDiff>;

export declare type BettererFileTestFunction = (filePaths: BettererFilePaths, fileTestResult: BettererFileTestResult) => MaybeAsync<void>;

export declare type BettererFileTestResult = BettererResultValueComplex & {
    addFile(absolutePath: string, fileText: string): BettererFile;
    getIssues(absolutePath: string): BettererFileIssues;
};

export declare type BettererPrinter<SerialisedType> = (serialised: SerialisedType) => MaybeAsync<string>;

export declare type BettererReporter = {
    configError?(config: BettererConfigPartial, error: BettererError): Promise<void> | void;
    contextStart?(context: BettererContext, lifecycle: Promise<BettererSummaries>): Promise<void> | void;
    contextEnd?(context: BettererContext, summary: BettererSummaries): Promise<void> | void;
    contextError?(context: BettererContext, error: BettererError): Promise<void> | void;
    runsStart?(runs: BettererRuns, files: BettererFilePaths): Promise<void> | void;
    runsEnd?(summary: BettererSummary, files: BettererFilePaths): Promise<void> | void;
    runStart?(run: BettererRun, lifecycle: Promise<void>): Promise<void> | void;
    runEnd?(run: BettererRun): Promise<void> | void;
    runError?(run: BettererRun, error: BettererError): Promise<void> | void;
};

export declare type BettererResult = {
    isNew: boolean;
    value: unknown;
    result: BettererResultValue;
};

export declare type BettererResultValue = BettererResultValueBasic | BettererResultValueComplex;

export declare type BettererResultValueBasic = number;

export declare type BettererResultValueComplex = {
    value: number;
};

export declare type BettererRun = {
    readonly diff: BettererDiff;
    readonly expected: BettererResult;
    readonly filePaths: BettererFilePaths;
    readonly lifecycle: Promise<void>;
    readonly name: string;
    readonly result: BettererResult;
    readonly test: BettererTestConfig;
    readonly timestamp: number;
    readonly isBetter: boolean;
    readonly isComplete: boolean;
    readonly isExpired: boolean;
    readonly isFailed: boolean;
    readonly isNew: boolean;
    readonly isObsolete: boolean;
    readonly isSame: boolean;
    readonly isSkipped: boolean;
    readonly isUpdated: boolean;
    readonly isWorse: boolean;
};

export declare type BettererRunNames = Array<string>;

export declare type BettererRuns = ReadonlyArray<BettererRun>;

export declare type BettererSerialise<DeserialisedType extends BettererResultValue, SerialisedType> = (result: DeserialisedType) => SerialisedType;

export declare type BettererSerialiser<DeserialisedType extends BettererResultValue, SerialisedType = DeserialisedType> = {
    serialise: BettererSerialise<DeserialisedType, SerialisedType>;
    deserialise: BettererDeserialise<DeserialisedType, SerialisedType>;
};

export declare type BettererStartConfigPartial = BettererBaseConfigPartial & Partial<{
    allowDiff: boolean;
    update: boolean;
}>;

export declare type BettererSummaries = Array<BettererSummary>;

export declare type BettererSummary = {
    readonly runs: BettererRuns;
    readonly result: string;
    readonly expected: string | null;
    readonly hasDiff: boolean;
    readonly better: BettererRuns;
    readonly completed: BettererRuns;
    readonly expired: BettererRuns;
    readonly failed: BettererRuns;
    readonly new: BettererRuns;
    readonly obsolete: BettererRuns;
    readonly ran: BettererRuns;
    readonly same: BettererRuns;
    readonly skipped: BettererRuns;
    readonly updated: BettererRuns;
    readonly worse: BettererRuns;
};

export declare class BettererTest<DeserialisedType extends BettererResultValue, SerialisedType, DiffType> implements BettererTestBase<DeserialisedType, SerialisedType, DiffType> {
    get config(): BettererTestConfig<DeserialisedType, SerialisedType, DiffType>;
    readonly isBettererTest = "isBettererTest";
    get isOnly(): boolean;
    get isSkipped(): boolean;
    constructor(config: BettererTestConfigPartial<DeserialisedType, SerialisedType, DiffType>);
    only(): this;
    skip(): this;
}

export declare type BettererTestConfig<DeserialisedType extends BettererResultValue = BettererResultValue, SerialisedType = DeserialisedType, DiffType = null> = {
    constraint: BettererTestConstraint<DeserialisedType>;
    deadline: number;
    goal: BettererTestGoal<DeserialisedType>;
    test: BettererTestFunction<DeserialisedType>;
    differ: BettererDiffer<DeserialisedType, DiffType>;
    printer: BettererPrinter<SerialisedType>;
    serialiser: BettererSerialiser<DeserialisedType, SerialisedType>;
};

export declare type BettererTestConstraint<DeserialisedType extends BettererResultValue> = (result: DeserialisedType, expected: DeserialisedType) => MaybeAsync<BettererConstraintResult>;

export declare type BettererTestFunction<DeserialisedType extends BettererResultValue> = (run: BettererRun) => MaybeAsync<DeserialisedType>;

export declare type BettererTestGoal<DeserialisedType extends BettererResultValue> = (result: DeserialisedType) => MaybeAsync<boolean>;

export declare type BettererWatchConfigPartial = BettererBaseConfigPartial & Partial<{
    ignores: BettererConfigIgnores;
}>;

export declare type BettererWatcher = {
    stop(): Promise<void>;
};

export declare function file(filePath: string, partialConfig?: BettererBaseConfigPartial): Promise<BettererSummary>;

export declare function watch(partialConfig?: BettererWatchConfigPartial): Promise<BettererWatcher>;
