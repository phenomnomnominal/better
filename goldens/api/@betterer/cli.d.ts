export declare type BettererCLIArguments = Array<string>;

export declare type BettererCLICIConfig = {
    config: BettererCLIArguments;
    filter: BettererCLIArguments;
    reporter: BettererCLIArguments;
    results: string;
    silent: boolean;
    tsconfig: string;
};

export declare type BettererCLIInitConfig = {
    config: string;
};

export declare type BettererCLIStartConfig = {
    config: BettererCLIArguments;
    filter: BettererCLIArguments;
    reporter: BettererCLIArguments;
    results: string;
    silent: boolean;
    tsconfig: string;
    update: boolean;
};

export declare type BettererCLIWatchConfig = BettererCLIStartConfig & {
    ignore: BettererCLIArguments;
};

export declare type BettererPackageJSON = {
    version: string;
    scripts: Record<string, string> & {
        betterer: string;
    };
    devDependencies: Record<string, string>;
};

export declare function ciΔ(cwd: string, argv: BettererCLIArguments): Promise<BettererSummary>;

export declare function cliΔ(argv: BettererCLIArguments): void;

export declare function initΔ(cwd: string, argv: BettererCLIArguments): Promise<void>;

export declare function startΔ(cwd: string, argv: BettererCLIArguments): Promise<BettererSummary>;

export declare function watchΔ(cwd: string, argv: BettererCLIArguments): Promise<void>;
