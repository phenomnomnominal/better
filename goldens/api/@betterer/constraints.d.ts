export declare enum BettererConstraintResult {
    better = "better",
    same = "same",
    worse = "worse"
}

export declare function bigger(result: number, expected: number): BettererConstraintResult;

export declare function smaller(result: number, expected: number): BettererConstraintResult;
