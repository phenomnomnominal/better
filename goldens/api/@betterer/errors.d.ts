export declare class BettererError extends Error {
    details: BettererErrorDetails;
    isBettererError: boolean;
    constructor(message: string, ...details: BettererErrorDetails);
}

export declare type BettererErrorDetail = string | Error | BettererError;

export declare type BettererErrorDetails = ReadonlyArray<BettererErrorDetail>;

export declare function isBettererError(err: unknown): err is BettererError;
