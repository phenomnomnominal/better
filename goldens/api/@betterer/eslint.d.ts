export declare function eslint(rules: ESLintRulesConfig): BettererFileTest;
