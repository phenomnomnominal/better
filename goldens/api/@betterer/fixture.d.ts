export declare function createFixtureDirectoryÎ”(fixturesPath: string): Promise<FixtureFactory>;

export declare type Fixture = FixtureFileSystem & {
    logs: ReadonlyArray<string>;
    waitForRun(watcher: BettererWatcher): Promise<BettererSummary>;
    runNames(runs: BettererRuns): BettererRunNames;
};

export declare type FixtureFactory = (fixtureName: string, files: FixtureFileSystemFiles, options?: FixtureOptions) => Promise<Fixture>;

export declare type FixtureFileSystem = {
    paths: Paths;
    deleteDirectory(filePath: string): Promise<void>;
    deleteFile(filePath: string): Promise<void>;
    readFile(filePath: string): Promise<string>;
    resolve(filePath: string): string;
    writeFile(filePath: string, text: string): Promise<void>;
    cleanup(): Promise<void>;
};

export declare type FixtureFileSystemFiles = Record<string, string>;

export declare type FixtureOptions = {
    logFilters?: Array<RegExp>;
};

export declare type Paths = {
    config: string;
    results: string;
    cwd: string;
};
