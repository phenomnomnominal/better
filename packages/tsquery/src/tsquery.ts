import { BettererFileResolver, BettererFileTest } from '@betterer/betterer';
import { BettererError } from '@betterer/errors';
import { tsquery as tsq } from '@phenomnomnominal/tsquery';
import { promises as fs } from 'fs';

export function tsquery(configFilePath: string, query: string): BettererFileTest {
  if (!configFilePath) {
    throw new BettererError(
      "for `@betterer/tsquery` to work, you need to provide the path to a tsconfig.json file, e.g. `'./tsconfig.json'`. ❌"
    );
  }
  if (!query) {
    throw new BettererError(
      "for `@betterer/tsquery` to work, you need to provide a query, e.g. `'CallExpression > PropertyAccessExpression'`. ❌"
    );
  }

  const resolver = new BettererFileResolver();
  const absoluteConfigFilePath = resolver.resolve(configFilePath);

  return new BettererFileTest(resolver, async (_, fileTestResult) => {
    const projectFiles = await resolver.validate(tsq.projectFiles(absoluteConfigFilePath));
    await Promise.all(
      projectFiles.map(async (filePath) => {
        const fileText = await fs.readFile(filePath, 'utf8');
        const sourceFile = tsq.ast(fileText);
        const file = fileTestResult.addFile(filePath, fileText);
        tsq.query(sourceFile, query, { visitAllChildren: true }).forEach((match) => {
          file.addIssue(match.getStart(), match.getEnd(), 'TSQuery match');
        });
      })
    );
  });
}
