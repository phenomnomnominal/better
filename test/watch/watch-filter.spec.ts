import type { BettererSuiteSummary } from '@betterer/betterer';

import { describe, expect, it } from 'vitest';

import { createFixture } from '../fixture.js';

describe('betterer.watch', () => {
  it.skipIf(!process.stdout.isTTY)('should filter based on input in watch mode', async () => {
    const { betterer } = await import('@betterer/betterer');

    const { logs, paths, resolve, cleanup, writeFile, sendKeys } = await createFixture('watch-filter', {
      '.betterer.ts': `
import { tsquery } from '@betterer/tsquery';

export default {
  test1: async () => {
    await new Promise(resolve => setTimeout(resolve, 2000));
    return tsquery(
      'CallExpression > PropertyAccessExpression[expression.name="console"][name.name="log"]'
    ).include('./src/**/*.ts')
  },
  test2: () => tsquery('DebuggerStatement').include('./src/**/*.ts')
};
      `
    });

    const configPaths = [paths.config];
    const resultsPath = paths.results;
    const indexPath = resolve('./src/index.ts');
    const { cwd } = paths;

    await writeFile(indexPath, `console.log('foo');`);

    await betterer({ configPaths, resultsPath, cwd, workers: false });

    const suiteSummaryDefers = [
      Promise.withResolvers<BettererSuiteSummary>(),
      Promise.withResolvers<BettererSuiteSummary>(),
      Promise.withResolvers<BettererSuiteSummary>()
    ];
    const [worse, filtered, worseFiltered] = suiteSummaryDefers;

    const runner = await betterer.watch({
      configPaths,
      resultsPath,
      cwd,
      reporters: [
        '@betterer/reporter',
        {
          suiteEnd(suiteSummary: BettererSuiteSummary) {
            const suiteSummaryDefer = suiteSummaryDefers.shift();
            suiteSummaryDefer?.resolve(suiteSummary);
          },
          suiteError(_, error) {
            const suiteSummaryDefer = suiteSummaryDefers.shift();
            suiteSummaryDefer?.reject(error);
          }
        }
      ],
      strict: false,
      workers: false
    });

    await writeFile(indexPath, `console.log('foo');debugger;`);

    const worseSuiteSummary = await worse.promise;

    expect(worseSuiteSummary.runSummaries).toHaveLength(2);

    // Press "f" to enter filter mode:
    await sendKeys('f');

    // Press "2" to target "test2":
    await sendKeys('2');

    // Press "Enter" to enable the filter:
    await sendKeys('\r');

    const filteredSuiteSummary = await filtered.promise;
    expect(filteredSuiteSummary.runSummaries).toHaveLength(2);

    await writeFile(indexPath, `console.log('foo');debugger;debugger;`);

    const filteredWorseSuiteSummary = await worseFiltered.promise;
    expect(filteredWorseSuiteSummary.runSummaries).toHaveLength(2);

    await runner.stop();

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
