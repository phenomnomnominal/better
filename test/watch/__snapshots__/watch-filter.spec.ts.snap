// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`betterer.watch > should filter based on input in watch mode 1`] = `
[
  "🌟 Betterer (0ms):
",
  "🌟 Betterer (0ms): 1 test running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!
",
  "🌟 Betterer (0ms): 2 tests running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!
",
  "🌟 Betterer (0ms): 1 test running... 1 test done!
✅ test1: "test1" got checked for the first time! (1 issue) 🎉
✅ test2: "test2" met its goal! 🎉
",
  "🎉 Betterer (0ms): 2 tests done!
✅ test1: "test1" got checked for the first time! (1 issue) 🎉
✅ test2: "test2" met its goal! 🎉
",
  "🎉 Betterer (0ms): 2 tests done!
✅ test1: "test1" got checked for the first time! (1 issue) 🎉
✅ test2: "test2" met its goal! 🎉

2 tests got checked. 🤔
1 test got checked for the first time! 🎉
1 test met its goal! 🎉
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Starting Betterer in watch mode! 🎉

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms):

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 2 tests running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running... 1 test done!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): No current filter patterns
Ignores (press "i" to edit): No current ignore patterns

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

2 tests got checked. 🤔
1 test stayed the same. 😐
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" or "q" to quit)
",
  "Press "enter" to confirm, or "esc" to cancel.
Use RegExp patterns e.g. /my test/. Use "," to separate multiple filters.
Filters:

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

2 tests got checked. 🤔
1 test stayed the same. 😐
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" to quit)
",
  "Press "enter" to confirm, or "esc" to cancel.
Use RegExp patterns e.g. /my test/. Use "," to separate multiple filters.
Filters: 2

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

2 tests got checked. 🤔
1 test stayed the same. 😐
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" stayed the same. (1 issue) 😐
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

2 tests got checked. 🤔
1 test stayed the same. 😐
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms):

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 2 tests running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running... 1 test done!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (1 new issue, 0 existing, 1 total) 😔
・ 1 new issue in "fixtures/watch-filter/src/index.ts".
・
・ > 1 | console.log('foo');debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

1 test got checked. 🤔
1 test got skipped. 🚫
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms):

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 2 tests running...
🤔 test1: running "test1"!
🤔 test2: running "test2"!

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

🌟 Betterer (0ms): 1 test running... 1 test done!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (2 new issues, 0 existing, 2 total) 😔
・ 2 new issues in "fixtures/watch-filter/src/index.ts".
・ Showing first of 2 new issues:
・
・ > 1 | console.log('foo');debugger;debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checking 1 file... 🤔

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (2 new issues, 0 existing, 2 total) 😔
・ 2 new issues in "fixtures/watch-filter/src/index.ts".
・ Showing first of 2 new issues:
・
・ > 1 | console.log('foo');debugger;debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

(press "ctrl+c" or "q" to quit)
",
  "Filters (press "f" to edit): 2
Ignores (press "i" to edit): No current ignore patterns

Checked 1 file! 🔍

・ fixtures/watch-filter/src/index.ts

💥 Betterer (0ms): 2 tests done! 1 test errored!
✅ test1: "test1" got skipped. 🚫
🔥 test2: "test2" got worse. (2 new issues, 0 existing, 2 total) 😔
・ 2 new issues in "fixtures/watch-filter/src/index.ts".
・ Showing first of 2 new issues:
・
・ > 1 | console.log('foo');debugger;debugger;
・     |                    ^^^^^^^^^ TSQuery match
・

1 test got checked. 🤔
1 test got skipped. 🚫
1 test got worse. 😔

You should try to fix the new issues! As a last resort, you can run \`betterer --update\` to force an update of the results file. 🆙

(press "ctrl+c" or "q" to quit)
",
  "Stopping watch mode... 👋
",
]
`;
